File: backend/src/service/comment.py
from flask import jsonify

from ..util.util import generate_uuid, get_now, utf8len
from ..database.database import db


def get_comments_of_related_uuid(related_uuid):
    data = db.query(
        "select * from comment where related_uuid='{}';".format(related_uuid))
    data.sort(key=lambda x: x['create_time'], reverse=True)
    if data is None:
        data = []
    return jsonify(data)


def delete_comment(uuid):
    db.run_sql("delete from comment where uuid='{}'".format(uuid))
    return "success"


def create_comment(related_uuid, content):
    db.insert_comment(generate_uuid(), related_uuid, content)
    return "success"


def update_comment(uuid, newContent):
    db.run_sql("update comment set content='{}' where uuid='{}'".format(
        newContent, uuid))
    return "success"

File: frontend/src/services/comment.ts
import { axiosInstance } from './axios';

// 获取关联的评论列表
export const getCommentsByRelatedUUID = (related_uuid: string) => {
    return axiosInstance
        .get(`/api/comment/get/by_related_uuid?related_uuid=${related_uuid}`)
        .then((data: any) => {
            return data.data;
        });
};

// 创建评论
export const createComment = async (related_uuid: string, content: string) => {
    const data = await axiosInstance.post(`/api/comment/create`, {
        related_uuid,
        content,
    });
    return data.data;
};

// 修改评论
export const updateComment = async (uuid: string, newContent: string) => {
    const data = await axiosInstance.post(`/api/comment/update`, {
        uuid,
        newContent,
    });
    return data.data;
};

// 删除评论
export const deleteComment = (uuid: string) => {
    return axiosInstance.delete(`/api/comment/delete?uuid=${uuid}`).then((data: any) => {
        return data.data;
    });
};
